android/src/main/java/com/expensify/livemarkdown/MarkdownTextWatcher.java

```diff
@@ -22,23 +22,17 @@ public void beforeTextChanged(CharSequence s, int start, int count, int after) {
 
   @Override
   public void onTextChanged(CharSequence s, int start, int before, int count) {
-    if (mShouldSkip) {
-      return;
-    }
-    // Set the flag to indicate text is being changed
-    mShouldSkip = true;
+   
   }
 
   @Override
   public void afterTextChanged(Editable editable) {
-      if (!mShouldSkip) {
-          return;
-      }
-      
       if (editable instanceof SpannableStringBuilder) {
+          if (mPreviousEditable != null && mPreviousEditable.toString().equals(editable.toString())) {
+              return;
+          }
           mMarkdownUtils.applyMarkdownFormatting((SpannableStringBuilder) editable);
-      }
-      
-      // Reset the flag after formatting is applied
-      mShouldSkip = false;
-  }}
+          mPreviousEditable = new SpannableStringBuilder(editable);
```

Comment:
Do we need to store `mPreviousEditable` as a `SpannableStringBuilder`? We just use `mPreviousEditable.toString()` so maybe let's keep it as a string?

Reply:
That's make sense, please wait

Reply:
done